frontend/src/pages/modelServing/screens/projects/NIMServiceModal/ManageNIMServingModal.tsx

```diff
@@ -222,15 +234,32 @@ const DeployNIMServiceModal: React.FC<DeployNIMServiceModalProps> = ({
       submitServingRuntimeResources({ dryRun: true }),
       submitInferenceServiceResource({ dryRun: true }),
     ])
-      .then(() =>
-        Promise.all([
-          submitServingRuntimeResources({ dryRun: false }),
-          submitInferenceServiceResource({ dryRun: false }),
-          createNIMSecret(namespace, NIM_SECRET_NAME, false, false),
-          createNIMSecret(namespace, NIM_NGC_SECRET_NAME, true, false),
-          createNIMPVC(namespace, nimPVCName, pvcSize, false),
-        ]),
-      )
+      .then(async () => {
+        const promises: Promise<void>[] = [
+          submitServingRuntimeResources({ dryRun: false }).then(() => undefined),
+          submitInferenceServiceResource({ dryRun: false }).then(() => undefined),
+        ];
+        if (!editInfo) {
+          promises.push(
+            createNIMSecret(namespace, NIM_SECRET_NAME, false, false).then(() => undefined),
+            createNIMSecret(namespace, NIM_NGC_SECRET_NAME, true, false).then(() => undefined),
+            createNIMPVC(namespace, nimPVCName, pvcSize, false).then(() => undefined),
+          );
+        } else if (pvc && pvc.spec.resources.requests.storage !== pvcSize) {
+          const createData: CreatingStorageObject = {
```

Comment:
Nit: This is not "createData" anything -- you're doing an edit.

Reply:
Looks like edit is working fine -- I'm going to merge this with this weird variable name -- feel free to clean it up in a future PR.